model/map-file/src/main/java/org/keycloak/models/map/storage/file/yaml/YamlParser.java

```diff
@@ -0,0 +1,259 @@
+
+    private static final Logger LOG = Logger.getLogger(YamlParser.class);
+    public static final String ARRAY_CONTEXT = "$@[]@$";
+
+    private static final ScalarResolver RESOLVER = new JsonScalarResolver();
+    private final Parser parser;
+    private final BlockContextStack contextStack;
+
+    // Leverage SnakeYaml's translation of primitive values
+    private static final class MiniConstructor extends StandardConstructor {
+
+        public MiniConstructor() {
+            super(SETTINGS);
+        }
+
+        // This has been based on SnakeYaml's own org.snakeyaml.engine.v2.constructor.BaseConstructor.constructObjectNoCheck(Node node)
+        @SuppressWarnings(value = "unchecked")
+        public Object constructStandardJavaInstance(ScalarNode node) {
+            return findConstructorFor(node)
+                .map(constructor -> constructor.construct(node))
+                .orElseThrow(() -> new ConstructorException(null, Optional.empty(), "could not determine a constructor for the tag " + node.getTag(), node.getStartMark()));
+        }
+
+        public static final MiniConstructor INSTANCE = new MiniConstructor();
+    }
+
+    private static final LoadSettings SETTINGS = LoadSettings.builder()
+      .setAllowRecursiveKeys(false)
+      .setParseComments(false)
+      .build();
+
+    public static <E> E parse(Path path, BlockContext<E> initialContext) {
+        LOG.tracef("parse(%s,%s)%s", path, initialContext, getShortStackTrace());
+
+        Objects.requireNonNull(path, "Path invalid");
+        try (InputStream is = Files.newInputStream(path)) {
+            if (Files.size(path) == 0) {
+                return null;
+            }
+            Parser p = new ParserImpl(SETTINGS, new StreamReader(SETTINGS, new YamlUnicodeReader(is)));
+            return new YamlParser<>(p, initialContext).parse();
+        } catch (IOException ex) {
+            LOG.warn(ex);
+            return null;
+        }
+    }
+
+    protected YamlParser(Parser p, BlockContext<E> initialContext) {
+        this.parser = p;
+        this.contextStack = new BlockContextStack(initialContext);
+    }
+
+    @SuppressWarnings("unchecked")
+    protected <E> E parse() {
+        consumeEvent(Event.ID.StreamStart, "Expected a stream");
+
+        if (!parser.checkEvent(Event.ID.StreamEnd)) {
+            consumeEvent(Event.ID.DocumentStart, "Expected a document in the stream");
+            parseNode();
+            consumeEvent(Event.ID.DocumentEnd, "Expected a single document in the stream");
+        }
+
+        consumeEvent(Event.ID.StreamEnd, "Expected a single document in the stream");
+
+        return (E) contextStack.pop().getResult();
+    }
+
+    protected Object parseNode() {
+        if (parser.checkEvent(Event.ID.Alias)) {
+            throw new IllegalStateException("Aliases are not handled at this moment");
+        }
+        Event ev = parser.next();
+//        System.out.println("  Parsing " + ev);
```

Comment:
Do we want to keep the commented out lines in the file? There are multiple occurences.

Reply:
No, they will be removed